File: experiments/parrot/csharp/attempts/opus46high/2/ParrotTests.cs
using Xunit;

namespace Parrot;

public class ParrotTests
{
    [Fact]
    public void SpeedOfEuropeanParrot()
    {
        var parrot = new Parrot(ParrotTypeEnum.EUROPEAN, 0, 0, false);
        Assert.Equal(12.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfAfricanParrotWithOneCoconut()
    {
        var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 1, 0, false);
        Assert.Equal(3.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfAfricanParrotWithTwoCoconuts()
    {
        var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 2, 0, false);
        Assert.Equal(0.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfAfricanParrotWithNoCoconuts()
    {
        var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 0, 0, false);
        Assert.Equal(12.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfNorwegianBlueParrotNailed()
    {
        var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 0, true);
        Assert.Equal(0.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfNorwegianBlueParrotNailedWithVoltage()
    {
        var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 1.5, true);
        Assert.Equal(0.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfNorwegianBlueParrotNotNailed()
    {
        var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 1.5, false);
        Assert.Equal(18.0, parrot.GetSpeed());
    }

    [Fact]
    public void SpeedOfNorwegianBlueParrotNotNailedHighVoltage()
    {
        var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 4, false);
        Assert.Equal(24.0, parrot.GetSpeed());
    }

    [Fact]
    public void CryOfEuropeanParrot()
    {
        var parrot = new Parrot(ParrotTypeEnum.EUROPEAN, 0, 0, false);
        Assert.Equal("Sqoork!", parrot.GetCry());
    }

    [Fact]
    public void CryOfAfricanParrot()
    {
        var parrot = new Parrot(ParrotTypeEnum.AFRICAN, 2, 0, false);
        Assert.Equal("Sqaark!", parrot.GetCry());
    }

    [Fact]
    public void CryOfNorwegianBlueParrotHighVoltage()
    {
        var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 4, false);
        Assert.Equal("Bzzzzzz", parrot.GetCry());
    }

    [Fact]
    public void CryOfNorwegianBlueParrotNoVoltage()
    {
        var parrot = new Parrot(ParrotTypeEnum.NORWEGIAN_BLUE, 0, 0, false);
        Assert.Equal("...", parrot.GetCry());
    }
}

File: experiments/parrot/fsharp/attempts/opus46high/5/ParrotTests.fs
module ParrotTests

open Xunit
open FsUnit.Xunit

[<Fact>]
let ``speed of european parrot`` () =
    Parrot.speed Parrot.European 0 0.0 false |> should equal 12.0

[<Fact>]
let ``speed of african parrot with one coconut`` () =
    Parrot.speed Parrot.African 1 0.0 false |> should equal 3.0

[<Fact>]
let ``speed of african parrot with two coconuts`` () =
    Parrot.speed Parrot.African 2 0.0 false |> should equal 0.0

[<Fact>]
let ``speed of african parrot with no coconuts`` () =
    Parrot.speed Parrot.African 0 0.0 false |> should equal 12.0

[<Fact>]
let ``speed of norwegian blue parrot nailed`` () =
    Parrot.speed Parrot.NorwegianBlue 0 0.0 true |> should equal 0.0

[<Fact>]
let ``speed of norwegian blue parrot nailed with voltage`` () =
    Parrot.speed Parrot.NorwegianBlue 0 1.5 true |> should equal 0.0

[<Fact>]
let ``speed of norwegian blue parrot not nailed`` () =
    Parrot.speed Parrot.NorwegianBlue 0 1.5 false |> should equal 18.0

[<Fact>]
let ``speed of norwegian blue parrot not nailed high voltage`` () =
    Parrot.speed Parrot.NorwegianBlue 0 4.0 false |> should equal 24.0

[<Fact>]
let ``cry of european parrot`` () =
    Parrot.cry Parrot.European 0.0 |> should equal "Sqoork!"

[<Fact>]
let ``cry of african parrot`` () =
    Parrot.cry Parrot.African 0.0 |> should equal "Sqaark!"

[<Fact>]
let ``cry of norwegian blue parrot high voltage`` () =
    Parrot.cry Parrot.NorwegianBlue 4.0 |> should equal "Bzzzzzz"

[<Fact>]
let ``cry of norwegian blue parrot no voltage`` () =
    Parrot.cry Parrot.NorwegianBlue 0.0 |> should equal "..."
